### ISSUE TITLE
`bosh create-env` fails with "No available SCSI Controller port\u0026device" error

### ISSUE THREAD

**Issue by ematpl** [cppforlife/bosh-virtualbox-cpi-release#1]
`bosh create-env` failed when trying to create a local BOSH-Lite using https://github.com/cloudfoundry/bosh-deployment.

Seems that nothing in the `showvminfo` output matched https://github.com/cppforlife/bosh-virtualbox-cpi-release/blob/v0.0.4/src/github.com/cppforlife/bosh-virtualbox-cpi/vm/port_devices.go#L14.

## System information

- OS: Mac OS X 10.11.6
- VirtualBox version: 5.1.10
- bosh-deployment at https://github.com/cloudfoundry/bosh-deployment/tree/594a008b0b026395f434d7d237974babd653bfcf

## Steps to reproduce

Updated VirtualBox CPI in bosh-deployment ops-file to v0.0.4 (but error also occurs with v0.0.3)

```
$ cd ~/workspace/bosh-deployment; git diff
diff --git a/virtualbox/cpi.yml b/virtualbox/cpi.yml
index 295b84a..618e7be 100644
--- a/virtualbox/cpi.yml
+++ b/virtualbox/cpi.yml
@@ -3,9 +3,9 @@
   path: /releases/-
   value:
     name: bosh-virtualbox-cpi
-    version: 0.0.3
-    url: https://bosh.io/d/github.com/cppforlife/bosh-virtualbox-cpi-release?v=0.0.3
-    sha1: f3d2c4d6e8cb5a7cd2c5c6e2a766de74c9b399ce
+    version: 0.0.4
+    url: https://bosh.io/d/github.com/cppforlife/bosh-virtualbox-cpi-release?v=0.0.4
+    sha1: 24c726b0a16f8114fd212596d3814f1303eb92d7

 - type: replace
   path: /resource_pools/name=vms/stemcell?
```

Ran `bosh create-env`:

```
$ bosh create-env \
  ~/workspace/bosh-deployment/bosh.yml \
  --state ./state.json \
  --ops-file ~/workspace/bosh-deployment/virtualbox/cpi.yml \
  --ops-file ~/workspace/bosh-deployment/bosh-lite.yml \
  --vars-store ./creds.yml \
  -v director_name=vbox \
  -v internal_ip=192.168.56.6 \
  -v internal_gw=192.168.56.1 \
  -v internal_cidr=192.168.56.0/24 \
  -v network_name=vboxnet0
Deployment manifest: '/Users/pivotal/workspace/bosh-deployment/bosh.yml'
Deployment state: './state.json'

Started validating
  Downloading release 'bosh'... Finished (00:00:25)
  Validating release 'bosh'... Finished (00:00:00)
  Downloading release 'bosh-virtualbox-cpi'... Finished (00:00:15)
  Validating release 'bosh-virtualbox-cpi'... Finished (00:00:00)
  Downloading release 'bosh-warden-cpi'... Finished (00:00:06)
  Validating release 'bosh-warden-cpi'... Finished (00:00:00)
  Downloading release 'garden-linux'... Finished (00:00:09)
  Validating release 'garden-linux'... Finished (00:00:00)
  Validating cpi release... Finished (00:00:00)
  Validating deployment manifest... Finished (00:00:00)
  Downloading stemcell... Finished (00:01:37)
  Validating stemcell... Finished (00:00:00)
Finished validating (00:02:37)

Started installing CPI
  Compiling package 'golang_1.7/21609f611781e8586e713cfd7ceb389cee429c5a'... Finished (00:01:40)
  Compiling package 'virtualbox_cpi/fc2707e787f8853acaea76f83010f1b53eba9af6'... Finished (00:00:16)
  Installing packages... Finished (00:00:02)
  Rendering job templates... Finished (00:00:00)
  Installing job 'virtualbox_cpi'... Finished (00:00:00)
Finished installing CPI (00:01:59)

Starting registry... Finished (00:00:00)
Uploading stemcell 'bosh-vsphere-esxi-ubuntu-trusty-go_agent/3312'... Finished (00:00:10)

Started deploying
  Creating VM for instance 'bosh/0' from stemcell 'sc-a8cb8e14-6250-4852-7c8e-0c32e7f43268'... Failed (00:00:05)
Failed deploying (00:00:05)

Stopping registry... Finished (00:00:00)
Cleaning up rendered CPI jobs... Finished (00:00:00)

Deploying:
  Creating instance 'bosh/0':
    Creating VM:
      Creating vm with stemcell cid 'sc-a8cb8e14-6250-4852-7c8e-0c32e7f43268':
        CPI 'create_vm' method responded with error: CmdError{"type":"Bosh::Clouds::CloudError","message":"Creating VM with agent ID '89edac2e-ad67-4e09-49be-0d3cf070eb72': Attaching ephemeral disk: No available SCSI Controller port\u0026device","ok_to_retry":false}

Exit code 1
```

## Diagnostic info:

```
$ VBoxManage showvminfo sc-a8cb8e14-6250-4852-7c8e-0c32e7f43268 --machinereadable
name="sc-a8cb8e14-6250-4852-7c8e-0c32e7f43268"
groups="/"
ostype="Ubuntu (64-bit)"
UUID="4315d509-93f8-4559-8aba-a18c2bbebd8c"
CfgFile="/Users/pivotal/VirtualBox VMs/sc-a8cb8e14-6250-4852-7c8e-0c32e7f43268/sc-a8cb8e14-6250-4852-7c8e-0c32e7f43268.vbox"
SnapFldr="/Users/pivotal/VirtualBox VMs/sc-a8cb8e14-6250-4852-7c8e-0c32e7f43268/Snapshots"
LogFldr="/Users/pivotal/VirtualBox VMs/sc-a8cb8e14-6250-4852-7c8e-0c32e7f43268/Logs"
hardwareuuid="4315d509-93f8-4559-8aba-a18c2bbebd8c"
memory=512
pagefusion="off"
vram=16
cpuexecutioncap=100
hpet="off"
chipset="piix3"
firmware="BIOS"
cpus=1
pae="on"
longmode="on"
triplefaultreset="off"
apic="on"
x2apic="on"
cpuid-portability-level=0
bootmenu="messageandmenu"
boot1="floppy"
boot2="dvd"
boot3="disk"
boot4="none"
acpi="on"
ioapic="on"
biosapic="apic"
biossystemtimeoffset=0
rtcuseutc="off"
hwvirtex="on"
nestedpaging="on"
largepages="on"
vtxvpid="on"
vtxux="on"
paravirtprovider="default"
effparavirtprovider="kvm"
VMState="poweroff"
VMStateChangeTime="2016-12-19T00:41:54.788000000"
monitorcount=1
accelerate3d="off"
accelerate2dvideo="off"
teleporterenabled="off"
teleporterport=0
teleporteraddress=""
teleporterpassword=""
tracing-enabled="off"
tracing-allow-vm-access="off"
tracing-config=""
autostart-enabled="off"
autostart-delay=0
defaultfrontend=""
storagecontrollername0="IDE"
storagecontrollertype0="PIIX4"
storagecontrollerinstance0="0"
storagecontrollermaxportcount0="2"
storagecontrollerportcount0="2"
storagecontrollerbootable0="on"
storagecontrollername1="SCSI"
storagecontrollertype1="LsiLogic"
storagecontrollerinstance1="0"
storagecontrollermaxportcount1="16"
storagecontrollerportcount1="16"
storagecontrollerbootable1="on"
"IDE-0-0"="none"
"IDE-0-1"="none"
"IDE-1-0"="none"
"IDE-1-1"="none"
"SCSI-0-0"="/Users/pivotal/VirtualBox VMs/sc-a8cb8e14-6250-4852-7c8e-0c32e7f43268/Snapshots/{85c94468-fa8a-44b8-ad9a-b8eb0e28305e}.vmdk"
"SCSI-ImageUUID-0-0"="85c94468-fa8a-44b8-ad9a-b8eb0e28305e"
"SCSI-1-0"="none"
"SCSI-2-0"="none"
"SCSI-3-0"="none"
"SCSI-4-0"="none"
"SCSI-5-0"="none"
"SCSI-6-0"="none"
"SCSI-7-0"="none"
"SCSI-8-0"="none"
"SCSI-9-0"="none"
"SCSI-10-0"="none"
"SCSI-11-0"="none"
"SCSI-12-0"="none"
"SCSI-13-0"="none"
"SCSI-14-0"="none"
"SCSI-15-0"="none"
nic1="none"
nic2="none"
nic3="none"
nic4="none"
nic5="none"
nic6="none"
nic7="none"
nic8="none"
hidpointing="ps2mouse"
hidkeyboard="ps2kbd"
uart1="off"
uart2="off"
uart3="off"
uart4="off"
lpt1="off"
lpt2="off"
audio="coreaudio"
clipboard="disabled"
draganddrop="disabled"
vrde="off"
usb="off"
ehci="off"
xhci="off"
vcpenabled="off"
vcpscreens=0
vcpfile="/Users/pivotal/VirtualBox VMs/sc-a8cb8e14-6250-4852-7c8e-0c32e7f43268/sc-a8cb8e14-6250-4852-7c8e-0c32e7f43268.webm"
vcpwidth=1024
vcpheight=768
vcprate=512
vcpfps=25
GuestMemoryBalloon=0
SnapshotName="prepared-clone"
SnapshotUUID="52e49c11-2735-47a3-9bcc-72f81994941d"
CurrentSnapshotName="prepared-clone"
CurrentSnapshotUUID="52e49c11-2735-47a3-9bcc-72f81994941d"
CurrentSnapshotNode="SnapshotName"
```

**Comment by cppforlife** [comment]
@ematpl looks like some time SCSI controllers got renamed from 

```
"SCSI Controller-1-0"="none"
"SCSI Controller-2-0"="none"
```

to just 

```
"SCSI-1-0"="none"
"SCSI-2-0"="none"
```

**Comment by cppforlife** [comment]
Hmm, looks like it depends on "storagecontrollername1" which for some reason is set to "SCSI Controller" on my box and "SCSI" on yours.

**Comment by cppforlife** [comment]
@ematpl Published v0.0.5 and bumped https://github.com/cloudfoundry/bosh-deployment.

**Comment by ematpl** [comment]
Awesome, thanks! Updated and verified that it's fixed for my VirtualBox configuration (however it ended up that way).
